Refactor Modules for Inbound Authentication (#15435)
ballerina-platform/ballerina-lang: stdlib/auth/src/main/ballerina/auth/Module.md

## Module Overview

This module provides default authentication provider configurations that can be extended to create new authentication providers.

### Authentication Provider

An authentication provider defines an authentication scheme that could be used to authenticate endpoints. The `auth:AuthStoreProvider` acts as the interface for all the authentication providers. Any type of implementation such as LDAP, JDBC, and file-based should be object-equivalent.

> When creating a new authentication provider, there is a function that needs to be implemented.

Reviewer: ```suggestion
When creating a new authentication provider, you need to implement the below function.
```

> - `authenticate` : Authenticates the user based on a credential, which can be username/password, or a token such as JWT.

Reviewer: ```suggestion
- `authenticate` : Authenticates the user based on the user credentials (i.e., the username/password) or a token such as JWT.
```

> By default, there is `auth:ConfigAuthStoreProvider`, which authenticates based on usernames and passwords stored in a configuration file. The `auth:ConfigAuthStoreProvider` is an implementation of the `auth:AuthProvider` interface, which uses the Ballerina configuration file to store usernames, passwords, scopes, and relevant associations. A user is denoted by a section in the configuration file. The password and the scopes assigned to the user are denoted as keys under the relevant user section as shown below.

Reviewer: ```suggestion
By default, the`auth:ConfigAuthStoreProvider` authenticates based on usernames and passwords stored in a configuration file. The `auth:ConfigAuthStoreProvider` is an implementation of the `auth:AuthProvider` interface, which uses the Ballerina configuration file to store usernames, passwords, scopes, and relevant associations.

A user is denoted by a section in the configuration file. The password and the scopes assigned to the user are denoted as keys under the relevant user section as shown below.
```
